packages/babel-plugin-transform-function-sent/src/index.js

```diff
@@ -0,0 +1,47 @@
+import syntaxFunctionSent from "babel-plugin-syntax-function-sent";
+import wrapFunction from "babel-helper-wrap-function";
+
+export default function({ types: t }) {
+  return {
+    inherits: syntaxFunctionSent,
+
+    visitor: {
+      MetaProperty(path) {
+        if (!t.isIdentifier(path.node.meta, { name: "function" })) return;
+        if (!t.isIdentifier(path.node.property, { name: "sent" })) return;
+
+        const fnPath = path.getFunctionParent();
+
+        if (!fnPath.node.generator) {
+          throw new Error("Parent generator function not found");
+        }
+
+        const sentId =
+          fnPath.getData("sentId") ||
+          fnPath.setData(
+            "sentId",
+            path.scope.generateUidIdentifier("function.sent"),
+          );
+
+        path.replaceWith(sentId);
+      },
+
+      Function: {
```

Comment:
I could not remove this visitor because I needed it to fix the `_functionSent` assignment (now it is updated every time `yield` is called).

Now this plugin works like this:
1. Replace `function.sent`s
2. Replace `yield`s (this needs a second traversal)
3. Wrap the function

I could not do 1 and 2 in a single step because if there is `yield` before `function.sent`, it is not possible to know if it must be replaced